internal/cmd/base/flags.go

```diff
@@ -682,6 +682,112 @@ func mapToKV(m map[string]string) string {
 	return strings.Join(list, ",")
 }
 
+// StringSliceMapVar maps a key string to a slice of string values.
+// This is useful for cases such as modifying Worker tags, which can have
+// multiple values associated per key.
+type StringSliceMapVar struct {
+	Name       string
+	Aliases    []string
+	Usage      string
+	Default    map[string][]string
+	Hidden     bool
+	Target     *map[string][]string
+	Completion complete.Predictor
+}
+
+func (f *FlagSet) StringSliceMapVar(i *StringSliceMapVar) {
+	f.VarFlag(&VarFlag{
+		Name:       i.Name,
+		Aliases:    i.Aliases,
+		Usage:      i.Usage,
+		Value:      newStringSliceMapValue(i.Default, i.Target, i.Hidden),
+		Completion: i.Completion,
+	})
+}
+
+type stringSliceMapValue struct {
+	hidden bool
+	target *map[string][]string
+}
+
+func newStringSliceMapValue(def map[string][]string, target *map[string][]string, hidden bool) *stringSliceMapValue {
+	*target = def
+	return &stringSliceMapValue{
+		hidden: hidden,
+		target: target,
+	}
+}
+
+func (s *stringSliceMapValue) Set(val string) error {
+	kv := strings.TrimSpace(val)
+	if kv == "null" {
+		s.target = nil
+		return nil
+	}
+
+	if *s.target == nil {
```

Comment:
This will panic if passed two values, where the first is `null`:

```
boundary workers set-worker-tags -id foobar -tag null -tag foo=bar
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x119b0a1]

goroutine 1 [running]:
github.com/hashicorp/boundary/internal/cmd/base.(*stringSliceMapValue).Set(0xc000c422b0, {0x7ffeb550c1f8, 0x7})
        /home/jeff/hc/boundary/internal/cmd/base/flags.go:728 +0xa1
flag.(*FlagSet).parseOne(0xc000bc8720)
        /usr/lib/go/src/flag/flag.go:1087 +0x643
flag.(*FlagSet).Parse(0xc000bc8720, {0xc00012e030?, 0xc0001a5a68?, 0x417f53?})
        /usr/lib/go/src/flag/flag.go:1106 +0x4f
github.com/hashicorp/boundary/internal/cmd/base.(*FlagSets).Parse(...)
        /home/jeff/hc/boundary/internal/cmd/base/base.go:495
github.com/hashicorp/boundary/internal/cmd/commands/workerscmd.(*Command).Run(0xc00069ef30, {0xc00012e030, 0x6, 0x6})
        /home/jeff/hc/boundary/internal/cmd/commands/workerscmd/workers.gen.go:136 +0x15c
github.com/mitchellh/cli.(*CLI).Run(0xc0004b2140)
        /home/jeff/go/pkg/mod/github.com/mitchellh/cli@v1.1.2/cli.go:262 +0x5f8
github.com/hashicorp/boundary/internal/cmd.RunCustom({0xc00012e010?, 0x60?, 0x0?}, 0x0?)
        /home/jeff/hc/boundary/internal/cmd/main.go:186 +0xa16
github.com/hashicorp/boundary/internal/cmd.Run(...)
        /home/jeff/hc/boundary/internal/cmd/main.go:92
main.main()
        /home/jeff/hc/boundary/cmd/boundary/main.go:13 +0xc7
```

The reason is you set the pointer to the map to `nil` above, but here you are checking if the map itself is nil and creating it if so. You can fix this either by adding an additional check on `s.target` or by setting `*s.target = nil` in the if clause above.

Reply:
thanks for the catch!